staffing/forms.py

```diff
@@ -221,51 +221,94 @@ def __init__(self, *args, **kwargs):
-            for day in days:
-                key = "charge_%s_%s" % (mission.id, day.day)
-                self.fields[key] = TimesheetFieldClass(required=False)
-                # Order tabindex by day
-                if day.isoweekday() in (6, 7) or day in holiday_days:
-                    tabIndex = 100000  # Skip week-end from tab path
-                    # Color week ends in grey
-                    self.fields[key].widget.attrs.setdefault("style", "background-color: LightGrey;")
-                else:
-                    tabIndex = day.day
-                self.fields[key].widget.attrs.setdefault("tabindex", tabIndex)
-
-                if day == days[0]:  # Only show label for first day
-                    tooltip = _("mission id: %s") % mission.mission_id()
-                    mission_link = escape(mission)
-                    if mission.lead_id:
-                        mission_link = "<a href='%s'>%s</a>" % (mission.get_absolute_url(), escape(mission))
-                    self.fields[key].label = mark_safe("<span class='pydici-tooltip' title='%s'>%s</span>" % (escape(tooltip), mission_link))
-                else:
-                    self.fields[key].label = ""
-            # Add staffing total and forecast in hidden field
-            hwidget = forms.HiddenInput()
-            # Mission id is added to ensure field key is uniq.
-            key = "%s %s %s" % (timesheetTotal.get(mission.id, 0), mission.id, forecastTotal[mission.id])
-            self.fields[key] = forms.CharField(widget=hwidget, required=False)
-
-        # Add lunch ticket line
-        if showLunchTickets:
-            for day in days:
-                key = "lunch_ticket_%s" % day.day
-                self.fields[key] = forms.BooleanField(required=False)
-                self.fields[key].widget.attrs.setdefault("size", 1)  # Reduce default size
-                self.fields[key].widget.attrs.setdefault("data-role", "none")  # Don't apply jquery theme
-                if day == days[0]:  # Only show label for first day
-                    self.fields[key].label = _("Days without lunch ticket")
-                else:
-                    self.fields[key].label = ""  # Squash label
-            # extra space is important - it is for forecast total (which does not exist for ticket...)
-            key = "%s total-ticket " % timesheetTotal.get("ticket", 0)
-            self.fields[key] = forms.CharField(widget=forms.HiddenInput(), required=False)
-
+        calendar_weeks = []
+        is_calendar_view = timesheet_view == 'calendar'
+        if is_calendar_view:
+            week = []
+            for idx, day in enumerate(days):
+                if day.isoweekday() == 1:
+                    week = []
+                week.append(day)
+                if day.isoweekday() == 7:
+                    calendar_weeks.append(week)
+
+        day_index = 0
+        for idxw, week_days in enumerate(calendar_weeks if is_calendar_view else [days]):
```

Comment:
We can reach 3 nested for, 3 if and again a for loop in the block... The form was already quite complicated, but handling both mode make is harder. I wonder if a real split in two distinct forms (but still compatible, in particular with the same field key, to work with the same view) may not be clearer ? To be discussed with a coffee next time you go to Paris :-)